### PR TITLE
fix: allow `DeliverMax` in `Payment` transaction

### PR DESCRIPTION
**Pull request by khancode**
## High Level Overview of Change

Reverts bug in rejecting `DeliverMax` param in `Payment` transaction

### Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### Did you update HISTORY.md?

- [ ] Yes
- [x] No, this change does not impact library users

## Test Plan

Added unit test.

### CODE REVIEW

File: packages/xrpl/src/models/transactions/payment.ts [hunk 1]
```diff
@@ -208,12 +208,6 @@ export function validatePayment(tx: Record<string, unknown>): void {
   }
 
   checkPartialPayment(tx)
-
```
Comment (ckeshava) [inline packages/xrpl/src/models/transactions/payment.ts:229]: Hello, I had worked on DeliverMax change in 2024. I had added a validation which compared the `Amount` and `DeliverMax` fields. If both of the fields were present and both had different values, we need to throw an error.

This validation appears to have been removed in the current version of the codebase.

Can you add this validation back in the code?

For further context, this PR builds upon this validation: https://github.com/XRPLF/xrpl.js/pull/2857

Comment (khancode) [reply packages/xrpl/src/models/transactions/payment.ts:229]: @ckeshava can you share the commit of when you added this validation?

**Review by Patel-Raj11 — APPROVED** [review]
LGTM!

**Comment by Patel-Raj11** [comment]
@mvadari Why did we add `DeliverMax` in payment transaction now? Was this to accommodate some changes for Batch transactions? Also, I don't see this in xrpl-py.

**Comment by mvadari** [comment]
> @mvadari Why did we add `DeliverMax` in payment transaction now? Was this to accommodate some changes for Batch transactions? Also, I don't see this in xrpl-py.

I didn't add it

**Comment by ckeshava** [comment]
@mvadari : I believe @Patel-Raj11  is asking why `DeliverMax` was included in [this PR change](https://github.com/XRPLF/xrpl.js/pull/2966). 

I wrote the changes pertaining to `DeliverMax` more than one year ago, [in this commit](https://github.com/XRPLF/xrpl.js/pull/2689).

File: packages/xrpl/src/models/transactions/payment.ts (hunk 1 above)
Comment (ckeshava) [reply packages/xrpl/src/models/transactions/payment.ts:229]: Hello Omar, this is the [PR for the original change](https://github.com/XRPLF/xrpl.js/pull/2689)

Comment (ckeshava) [reply packages/xrpl/src/models/transactions/payment.ts:229]: The integ-tests in this file evaluate those aspects of the validation: https://github.com/XRPLF/xrpl.js/pull/2689/files#diff-4fbead92a8c0e3529d7c7a8b04d552b22a79264ea9147c0f7e79bf2ee35e16a2

Comment (khancode) [reply packages/xrpl/src/models/transactions/payment.ts:229]: @ckeshava the validation is still here but it was refactored - https://github.com/XRPLF/xrpl.js/blob/2309d591ec0c07d40ed00de69f8490d033bbbbcd/packages/xrpl/src/client/index.ts#L694-L696

**Comment by mvadari** [comment]
> @mvadari : I believe @Patel-Raj11 is asking why `DeliverMax` was included in [this PR change](https://github.com/XRPLF/xrpl.js/pull/2966).

A mistake

File: packages/xrpl/src/models/transactions/payment.ts (hunk 1 above)
Comment (khancode) [reply packages/xrpl/src/models/transactions/payment.ts:229]: Integ tests are also still here.

Comment (ckeshava) [reply packages/xrpl/src/models/transactions/payment.ts:229]: @khancode  The refactor has two mistakes.
- The `handleDeliverMax` method is only invoked inside the [autofill-related function](https://github.com/XRPLF/xrpl.js/blob/2309d591ec0c07d40ed00de69f8490d033bbbbcd/packages/xrpl/src/client/index.ts#L695). Consequently, non-autofill code-paths will not validate the `DeliverMax` field at all. In the original implementation, this validation was performed inside the `Payment` transaction model.

More importantly, what is the benefit offered by this refactored? What is the goal of that PR change?

- In the `handleDeliverMax` method, [the `DeliverMax` field is deleted here](https://github.com/XRPLF/xrpl.js/blob/2309d591ec0c07d40ed00de69f8490d033bbbbcd/packages/xrpl/src/sugar/autofill.ts#L421). What is the need for removing the `DeliverMax` field? There is no harm in retaining it.

From an end-user's perspective, this causes confusion. When users look at the transaction response, it will not match their inputs while constructing the transaction object.

Comment (ckeshava) [reply packages/xrpl/src/models/transactions/payment.ts:229]: > Integ tests are also still here.

Did you look at [this branch of mine](https://github.com/ckeshava/xrpl.js/pull/new/checkoutDeliverMax)? I had referenced it in the context of the Jira issue. That test case in that branch is failing. Clearly, that cannot be present in the current code base. If it was, this error would have been caught by the CI.

Comment (ckeshava) [reply packages/xrpl/src/models/transactions/payment.ts:229]: Furthermore, the Payment integration tests are missing one corner case. This case was present in my original commit, however it appears to have been removed in recent history.

The below case needs to be included inside: `packages/xrpl/test/integration/transactions/payment.test.ts`

```
it(
    'Validate Payment transaction API v2: Payment Transaction: identical DeliverMax and Amount fields',
    async () => {
      paymentTx.DeliverMax = '2000'
      paymentTx.Amount = '1000'

      const result = await testTransaction(
        testContext.client,
        paymentTx,
        senderWallet,
      )

      // you need to assert for a validation-error
    },
    TIMEOUT,
  )
```

_Note: This test case is different from what I present in my custom branch. This code tests the `autofill` code-path, unlike the unit-test._

Please go through the original commit for details of the implementation.

File: packages/xrpl/src/models/transactions/payment.ts [hunk 2]
```diff
@@ -226,12 +226,6 @@ export function validatePayment(tx: Record<string, unknown>): void {
...
-  if (tx.DeliverMax != null) {
```
Comment (ckeshava) [inline packages/xrpl/src/models/transactions/payment.ts:230]: @khancode  You still need to invoke handleDeliverMax method at this line. In the present state, the `Payment` transaction model does not do any validation on `DeliverMax` at all.

File: packages/xrpl/src/models/transactions/payment.ts (hunk 1 above)
Comment (ckeshava) [reply packages/xrpl/src/models/transactions/payment.ts:229]: In the current PR, the deliver_max field is being validated inside the `autofill` method. However, if users do not make use of the autofill functionality, then the DeliverMax field will not be validated for compatibility.

In the interest of time/developer-effort, this is an acceptable compromise. If any users need/want to use the `Payment` transaction-validation without the autofill-functonality, please raise an issue.

**Review by ckeshava — APPROVED** [review]
